[CI] Reduce PGJournal.MarkTerminal cognitive complexity below lint threshold

## 现状

Develop CI is red across multiple runs because `full-ci / build-test (kernel, ./kernel/..., true)` runs the workspace satellite linter and fails in `./adapters/postgres`. The root module `golangci-lint` passes; the failing check is the satellite module lint.

First observed in run https://github.com/ghbvf/gocell/actions/runs/27469036405 on commit `673f1286` (`feat(saga): PG saga-journal global scan + migration + PG source + owner-CAS`). The same failure repeats in later develop pushes:

- https://github.com/ghbvf/gocell/actions/runs/27469878002
- https://github.com/ghbvf/gocell/actions/runs/27470477092
- https://github.com/ghbvf/gocell/actions/runs/27470488099
- https://github.com/ghbvf/gocell/actions/runs/27470573037

Representative log:

```text
+++ golangci-lint in module: ./adapters/postgres
adapters/postgres/saga/journal_store.go:517:1: cognitive complexity 16 of func `(*PGJournal).MarkTerminal` is high (> 15) (gocognit)
func (s *PGJournal) MarkTerminal(ctx context.Context, instanceID, leaseID idutil.SafeID, finalStatus saga.Status) (bool, error) {
^
1 issues:
* gocognit: 1
```

The line is `512` in the first failing run and `517` in later commits, but the function is the same: `(*PGJournal).MarkTerminal`.

## 修复方向

Refactor `(*PGJournal).MarkTerminal` so its cognitive complexity is <= 15 without changing saga journal semantics.

Likely minimal split:

- extract terminal status validation / terminal event kind resolution into a helper;
- extract transaction commit / rollback handling or event append transition logic into a helper;
- keep stale lease / missing instance behavior unchanged (`false, nil`);
- preserve the advisory-lock protocol around terminal event append.

Verify with:

```bash
go -C adapters/postgres run github.com/golangci/golangci-lint/v2/cmd/golangci-lint@v2.11.4 run ./...
# or the repo helper used by CI for workspace satellites
```

## Files

- adapters/postgres/saga/journal_store.go:517

## Trigger

N/A

## Source

CI investigation for runs:
- https://github.com/ghbvf/gocell/actions/runs/27469036405
- https://github.com/ghbvf/gocell/actions/runs/27469878002
- https://github.com/ghbvf/gocell/actions/runs/27470477092
- https://github.com/ghbvf/gocell/actions/runs/27470488099
- https://github.com/ghbvf/gocell/actions/runs/27470573037

--- a/adapters/postgres/saga/journal_store.go
+++ b/adapters/postgres/saga/journal_store.go
@@ -558,27 +558,8 @@ func (s *PGJournal) MarkTerminal(ctx context.Context, instanceID, leaseID idutil
 		return false, err
 	}
 
-	if _, err := tx.Exec(ctx, updateInstanceTerminal,
-		statusToInt16(finalStatus), now, string(instanceID), string(leaseID),
-	); err != nil {
-		return false, errcode.Wrap(errcode.KindInternal, kerrors.ErrAdapterPGQuery,
-			"saga journal: MarkTerminal update failed", err)
-	}
-
-	// Serialize saga_events INSERTs so global_seq order == commit order (F1).
-	// See sagaEventsGlobalAppendLockKey for the full rationale.
-	if _, err := tx.Exec(ctx, advisoryLockSQL, sagaEventsGlobalAppendLockKey); err != nil {
-		return false, errcode.Wrap(errcode.KindInternal, kerrors.ErrAdapterPGQuery,
-			"saga journal: MarkTerminal advisory lock", err)
-	}
-
-	version := row.currentVersion + 1
-	if _, err := tx.Exec(ctx, insertEvent,
-		string(instanceID), version, kindToInt16(termKind),
-		nullableStepName(""), nil, now,
-	); err != nil {
-		return false, errcode.Wrap(errcode.KindInternal, kerrors.ErrAdapterPGQuery,
-			"saga journal: MarkTerminal insert event failed", err)
+	if err := markTerminalWrites(ctx, tx, instanceID, leaseID, finalStatus, termKind, row.currentVersion+1, now); err != nil {
+		return false, err
 	}
 
 	if owned {
@@ -589,6 +570,39 @@ func (s *PGJournal) MarkTerminal(ctx context.Context, instanceID, leaseID idutil
 		committed = true
 	}
 	return true, nil
+}
+
+// markTerminalWrites performs the terminal projection update plus the
+// advisory-locked terminal event append inside the caller's tx. Split out of
+// MarkTerminal to keep that method under the cognitive-complexity budget; the
+// commit-order protocol (F1, #1630) is unchanged — sagaEventsGlobalAppendLockKey
+// is still acquired immediately before the insertEvent INSERT.
+func markTerminalWrites(
+	ctx context.Context, tx pgx.Tx, instanceID, leaseID idutil.SafeID,
+	finalStatus saga.Status, termKind journal.EventKind, version int64, now time.Time,
+) error {
+	if _, err := tx.Exec(ctx, updateInstanceTerminal,
+		statusToInt16(finalStatus), now, string(instanceID), string(leaseID),
+	); err != nil {
+		return errcode.Wrap(errcode.KindInternal, kerrors.ErrAdapterPGQuery,
+			"saga journal: MarkTerminal update failed", err)
+	}
+
+	// Serialize saga_events INSERTs so global_seq order == commit order (F1).
+	// See sagaEventsGlobalAppendLockKey for the full rationale.
+	if _, err := tx.Exec(ctx, advisoryLockSQL, sagaEventsGlobalAppendLockKey); err != nil {
+		return errcode.Wrap(errcode.KindInternal, kerrors.ErrAdapterPGQuery,
+			"saga journal: MarkTerminal advisory lock", err)
+	}
+
+	if _, err := tx.Exec(ctx, insertEvent,
+		string(instanceID), version, kindToInt16(termKind),
+		nullableStepName(""), nil, now,
+	); err != nil {
+		return errcode.Wrap(errcode.KindInternal, kerrors.ErrAdapterPGQuery,
+			"saga journal: MarkTerminal insert event failed", err)
+	}
+	return nil
 }
 
 // RepoReady probes the saga_instances relation so schema/migration drift

--- a/examples/ssobff/go.mod
+++ b/examples/ssobff/go.mod
@@ -13,7 +13,7 @@ require (
 	github.com/ghbvf/gocell/adapters/rabbitmq v0.0.0 // indirect
 	github.com/ghbvf/gocell/adapters/redis v0.0.0 // indirect
 	github.com/rabbitmq/amqp091-go v1.11.0 // indirect
-	github.com/redis/go-redis/v9 v9.20.0 // indirect
+	github.com/redis/go-redis/v9 v9.20.1 // indirect
 	github.com/santhosh-tekuri/jsonschema/v6 v6.0.2 // indirect
 	go.uber.org/atomic v1.11.0 // indirect
 )

